pkg/kubelet/allocation/allocation_manager_test.go

```diff
@@ -1020,6 +1020,271 @@ func TestHandlePodResourcesResizeWithSwap(t *testing.T) {
 	}
 }
 
+// testPodAdmitHandler is a lifecycle.PodAdmitHandler for testing.
+type testPodAdmitHandler struct {
+	// admitFunc contains the custom logic for admitting or rejecting a pod.
+	admitFunc func(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult
+}
+
+// Admit rejects all pods in the podsToReject list with a matching UID.
+func (a *testPodAdmitHandler) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
+	if a.admitFunc == nil {
+		return lifecycle.PodAdmitResult{Admit: true}
+	}
+	return a.admitFunc(attrs)
+}
+
+func TestAllocationManagerAddPod(t *testing.T) {
+	if goruntime.GOOS == "windows" {
+		t.Skip("InPlacePodVerticalScaling is not currently supported for Windows")
+	}
+	featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.InPlacePodVerticalScaling, true)
+
+	const containerName = "c1"
+
+	cpu1Mem1G := v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}
+	cpu2Mem2G := v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("2Gi")}
+
+	pod1UID := types.UID("111")
+	pod1Small := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{UID: pod1UID, Name: "pod1", Namespace: "ns1"},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  containerName,
+					Image: "i1",
+					Resources: v1.ResourceRequirements{
+						Requests: cpu1Mem1G,
+					},
+				},
+			},
+		},
+	}
+	pod1Large := pod1Small.DeepCopy()
+	pod1Large.Spec.Containers[0].Resources.Requests = cpu2Mem2G
+
+	pod2UID := types.UID("222")
+	pod2Small := &v1.Pod{
```

Comment:
nit (feel free to ignore): is `pod2Small` the same as `pod1Small` just with a different name/ns/uid? Consider making a common helper function that takes in those 3 fields as parameters to make this more obvious